<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>994</width>
    <height>669</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Random Walker Segmentation</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="1" rowspan="3">
     <widget class="TViewerWidget" name="viewer" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QGroupBox" name="group_summary">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>390</width>
        <height>0</height>
       </size>
      </property>
      <property name="title">
       <string>Summary</string>
      </property>
      <layout class="QFormLayout" name="formLayout">
       <item row="0" column="0">
        <widget class="QLabel" name="label_dataset_name">
         <property name="text">
          <string>Dataset name</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLabel" name="label_dataset_name_value">
         <property name="text">
          <string>-</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_num_points">
         <property name="text">
          <string>Number of points</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLabel" name="label_num_points_value">
         <property name="text">
          <string>-</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_graph">
         <property name="text">
          <string>Graph</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLabel" name="label_graph_value">
         <property name="text">
          <string>-</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_seeds">
         <property name="text">
          <string>Seeds</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QLabel" name="label_seeds_value">
         <property name="text">
          <string>-</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QGroupBox" name="group_pipeline">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>390</width>
        <height>0</height>
       </size>
      </property>
      <property name="title">
       <string>Pipeline</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QToolBox" name="toolbox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="currentIndex">
          <number>0</number>
         </property>
         <widget class="PreprocessingForm" name="page_preprocessing">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>366</width>
            <height>330</height>
           </rect>
          </property>
          <attribute name="label">
           <string>Preprocessing</string>
          </attribute>
         </widget>
         <widget class="TrunkDetectionForm" name="page_trunk_detection">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>100</width>
            <height>30</height>
           </rect>
          </property>
          <attribute name="label">
           <string>Trunk detection</string>
          </attribute>
         </widget>
         <widget class="GraphBuildingForm" name="page_graph_building">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>100</width>
            <height>30</height>
           </rect>
          </property>
          <attribute name="label">
           <string>Graph building</string>
          </attribute>
         </widget>
         <widget class="SegmentationForm" name="page_segmentation">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>100</width>
            <height>30</height>
           </rect>
          </property>
          <attribute name="label">
           <string>Segmentation</string>
          </attribute>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menu_bar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>994</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_file">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="action_save_seeds"/>
    <addaction name="action_load_seeds"/>
    <addaction name="separator"/>
    <addaction name="action_save_segmentation"/>
    <addaction name="separator"/>
    <addaction name="action_exit"/>
   </widget>
   <addaction name="menu_file"/>
  </widget>
  <widget class="QStatusBar" name="status_bar"/>
  <action name="action_save_viewpoint">
   <property name="text">
    <string>Save viewpoint</string>
   </property>
  </action>
  <action name="action_load_viewpoint">
   <property name="text">
    <string>Load viewpoint</string>
   </property>
  </action>
  <action name="action_save_seeds">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Save seeds</string>
   </property>
  </action>
  <action name="action_load_seeds">
   <property name="text">
    <string>Load seeds</string>
   </property>
  </action>
  <action name="action_exit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+E</string>
   </property>
   <property name="menuRole">
    <enum>QAction::QuitRole</enum>
   </property>
  </action>
  <action name="action_save_segmentation">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Save segmentation</string>
   </property>
  </action>
  <action name="action_highlight_cluster_mode">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Highlight cluster mode</string>
   </property>
   <property name="shortcut">
    <string>H</string>
   </property>
  </action>
  <action name="action_unique_labels">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Unique labels for &quot;no-seeds&quot; clusters</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>TViewerWidget</class>
   <extends>QWidget</extends>
   <header>tviewer/tviewer_widget/tviewer_widget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>PreprocessingForm</class>
   <extends>QWidget</extends>
   <header>preprocessing_form.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>TrunkDetectionForm</class>
   <extends>QWidget</extends>
   <header>trunk_detection_form.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>GraphBuildingForm</class>
   <extends>QWidget</extends>
   <header>graph_building_form.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>SegmentationForm</class>
   <extends>QWidget</extends>
   <header>segmentation_form.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>action_exit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onActionExitTriggered()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_load_seeds</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onActionLoadSeedsTriggered()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_save_seeds</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onActionSaveSeedsTriggered()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_save_segmentation</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onActionSaveSegmentationTriggered()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>334</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toolbox</sender>
   <signal>currentChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>onCurrentToolboxPageChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>133</x>
     <y>378</y>
    </hint>
    <hint type="destinationlabel">
     <x>496</x>
     <y>334</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>buttonNewLabelClicked()</slot>
  <slot>buttonSegmentClicked()</slot>
  <slot>buttonDeleteLabelClicked()</slot>
  <slot>onActionExitTriggered()</slot>
  <slot>onActionLoadSeedsTriggered()</slot>
  <slot>onActionSaveSeedsTriggered()</slot>
  <slot>onButtonUpdateGraphClicked()</slot>
  <slot>onActionSaveSegmentationTriggered()</slot>
  <slot>onCurrentToolboxPageChanged(int)</slot>
 </slots>
</ui>
